libraries/lib-wave-track/WaveClip.cpp

```diff
@@ -1146,6 +1160,65 @@ void WaveClip::Resample(int rate, BasicUI::ProgressDialog *progress)
    }
 }
 
+void WaveClip::ApplyStretchRatio()
+{
+   const auto stretchRatio = GetStretchRatio();
+   if (stretchRatio == 1.0)
+      return;
+
+   Finally Do { [this] {
+      this->mClipStretchRatio = 1.0;
+      this->mRawAudioTempo = this->mProjectTempo;
+   } };
+
+   constexpr auto durationToDiscard = 0.0;
+   constexpr auto blockSize = 1024;
+   const auto numChannels = GetWidth();
+
+   // Let's also pass the hidden part to the stretcher for better result in the
+   // visible part.
+   const auto trimLeftBeforeStretch = mTrimLeft;
+   const auto trimRightBeforeStretch = mTrimRight;
+   SetTrimLeft(0);
+   SetTrimRight(0);
+
+   ClipSegment stretcherSource { *this, durationToDiscard,
+                                 PlaybackDirection::forward };
+   TimeAndPitchInterface::Parameters params;
+   params.timeRatio = stretchRatio;
+   StaffPadTimeAndPitch stretcher { numChannels, stretcherSource,
+                                    std::move(params) };
+   const auto totalNumOutSamples =
+      sampleCount { GetVisibleSampleCount().as_double() * stretchRatio + .5 };
+   sampleCount numProcessedSamples { 0 };
+
+   auto newSequences = GetEmptySequenceCopies();
+
+   sampleCount numOutSamples { 0 };
+   AudioContainer container(blockSize, numChannels);
+   while (numOutSamples < totalNumOutSamples)
+   {
+      const auto numSamplesToGet = limitSampleBufferSize(
+         blockSize, totalNumOutSamples - numProcessedSamples);
```

Comment:
`numProcessedSamples` never varies.  Should you remove it and use `numOutSamples`?

Reply:
Oj - thanks. This was a bug, likely appending unwanted silence samples.